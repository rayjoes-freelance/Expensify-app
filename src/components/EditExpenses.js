import React from 'react';
import {connect} from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const EditExpensePage = (props) => {
    console.log(props.expense.id);
    return (
        <div>
            <p>Editing the expense with the id of {props.match.params.id}</p>
            <ExpenseForm
            expense={props.expense}
            onSubmit={(expense) => {
                props.dispatch(editExpense(props.expense.id,expense));
                props.history.push('/');
                }
            }
            />
            <button onClick={() => {
                props.dispatch(removeExpense(props.expense.id));
                props.history.push('/');
            }}>
                Remove
            </button>
        </div>
    );
};

const mapStateToProps = (state, props) => {
    console.log(state.expenses.find((expense) => expense.id === props.match.params.id));
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};

export default connect(mapStateToProps)(EditExpensePage);