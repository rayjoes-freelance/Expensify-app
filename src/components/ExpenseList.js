import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpense from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
    <h1>Expense List</h1>
    {
        props.expenses == undefined ? (
            <p> there is currently no expenses</p>
        ) : 
        (
            props.expenses.map((expense) => {
            return <ExpenseListItem {...expense} key={expense.id}/>;
        })
        )
    }   
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses : selectExpense(state.expenses, state.filters),

    };
};


export default connect(mapStateToProps)(ExpenseList);