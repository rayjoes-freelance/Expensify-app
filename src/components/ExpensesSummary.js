import React from 'react';
import getExpenseTotal from '../selectors/expenses-total';
import numeral from 'numeral';


 export default (props) =>{
    console.log(props.expenses);
    let expensesNumber = props.expenses.length;
    let expensesTotal =numeral(getExpenseTotal(props.expenses)/ 100)
    .format('$0,0.00');
    return (
        <div>
        <h4>{`Viewing ${expensesNumber} expenses totalling of ${expensesTotal}`} </h4>
    </div>
    );
};
