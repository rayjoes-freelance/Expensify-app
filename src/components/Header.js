import React from 'react';
import {NavLink} from 'react-router-dom';


const Header = () => {
    return (
        <div>
        <header>
          <h1>Depensify</h1>
        </header>
        <ul>
            <li><NavLink exact={true} to="/" activeClassName="is-active">Home page</NavLink></li>
            <li><NavLink to="/create" activeClassName="is-active">Create Expenses</NavLink></li>
            <li><NavLink to="/help" activeClassName="is-active">Help Page </NavLink></li>
        </ul>
        </div>
    );
};

export default Header;