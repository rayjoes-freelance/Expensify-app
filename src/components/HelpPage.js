import React from 'react';

const HelpPage = () => {
    return (
        <div>
            This is help page 
        </div>
    );
};

export default HelpPage;