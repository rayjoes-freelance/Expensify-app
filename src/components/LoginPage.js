import React from 'react';


export default () => (
    <div>
        <h1>Login Page</h1>
        <form>
            <label><b>Username</b></label>
            <input type='texte' placeholder='Username' required/>
            <label><b>Password</b></label>
            <input type='password' placeholder='Password' required/>
            <button>Login</button>
        </form>
    </div>
);