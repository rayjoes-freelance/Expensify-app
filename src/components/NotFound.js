import React from 'react';
import {Link} from 'react-router-dom';

const NotFound = () => {
    return (
        <div>
            Error 404! <Link to="/">go home</Link>
        </div>
    );
};

export default NotFound;