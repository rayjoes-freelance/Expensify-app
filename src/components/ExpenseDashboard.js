import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
import selectExpense from '../selectors/expenses';
import {connect} from 'react-redux';

const ExpenseDashboardPage = (props) => {
    return (
        <div>
            <ExpensesSummary expenses={props.expenses}/>
            <ExpenseListFilters/>
            <ExpenseList/>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        expenses : selectExpense(state.expenses, state.filters),

    };
};


export default connect(mapStateToProps)(ExpenseDashboardPage);