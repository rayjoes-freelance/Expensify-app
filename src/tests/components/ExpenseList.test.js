import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expensesArray';

test('should render an expense list with expenses', () =>{
    const wrapper = shallow(<ExpenseList expenses= {expenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render expense list with an empty message', () => {
    const wrapper = shallow(<ExpenseList />);
    expect(wrapper).toMatchSnapshot();
});
