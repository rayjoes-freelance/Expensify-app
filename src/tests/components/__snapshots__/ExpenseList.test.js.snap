// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense list with an empty message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
     there is currently no expenses
  </p>
</div>
`;

exports[`should render an expense list with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={196}
    createAt={-100000}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createAt={345600000}
    description="credit-card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
